File: api/improve-resume.js
import { Configuration, OpenAIApi } from "openai";

const configuration = new Configuration({
    apiKey: process.env.OPENAI_API_KEY,
});

const openai = new OpenAIApi(configuration);

export default async function handler(req, res) {
    if (req.method !== 'POST') {
        res.setHeader('Allow', ['POST']);
        return res.status(405).end(`Method ${req.method} Not Allowed`);
    }

    const { resume_text } = req.body;

    if (!resume_text) {
        return res.status(400).json({ error: 'Resume text is required' });
    }

    try {
        const response = await openai.createCompletion({
            model: "text-davinci-003",
            prompt: `Improve this resume:\n\n${resume_text}`,
            max_tokens: 500,
        });

        res.status(200).json({ improved_resume: response.data.choices[0].text.trim() });
    } catch (error) {
        console.error('OpenAI API error:', error);
        res.status(500).json({ error: 'Failed to improve resume' });
    }
}

File: api/improve_resume.py
import openai
import os
from flask import Blueprint, request, jsonify

improve_resume_bp = Blueprint('improve_resume', __name__)

@improve_resume_bp.route('/api/improve-resume', methods=['POST'])
def improve_resume():
    try:
        data = request.get_json()
        resume_text = data['resume_text']

        # تنظیم کلید API از متغیر محیطی
        openai.api_key = os.getenv('OPENAI_API_KEY')
        
        response = openai.Chat.create(
            model="gpt-3.5-turbo",
            messages=[
                {"role": "system", "content": "You are a helpful assistant."},
                {"role": "user", "content": f"Improve the following resume:\n\n{resume_text}"}
            ],
            max_tokens=1000
        )

        improved_resume = response['choices'][0]['message']['content'].strip()
        return jsonify({'improved_resume': improved_resume})

    except Exception as e:
        return jsonify({'error': str(e)}), 500

